packages/date-picker/docs/DateRangeInputField.stories.tsx

```diff
@@ -127,15 +127,85 @@ const StickerSheetTemplate: Story<{
               locale={locale}
               disabled
             />
+          </StickerSheet.Row>
+        </StickerSheet.Body>
+      </StickerSheet>
+
+      <StickerSheet heading="Validation">
+        <StickerSheet.Header
+          headings={[
+            "Error - DateStart",
+            "Error - DateEnd",
+            "Error - Both",
+            "Different Status",
+          ]}
+        />
+        <StickerSheet.Body>
+          <StickerSheet.Row style={{ verticalAlign: "top" }}>
             <DateRangeInputField
-              id="daterangeinputfield--error"
+              id="daterangeinputfield--error-start"
               legend="Error example"
               isReversed={isReversed}
               inputRangeStartProps={inputRangeStartProps}
               inputRangeEndProps={inputRangeEndProps}
               locale={locale}
-              status="error"
-              validationMessage="Error message"
+              status={{
+                dateStart: "error",
+              }}
+              validationMessage={
+                '"Date from" cannot be after than the "Date to" selection.'
+              }
+            />
+            <DateRangeInputField
+              id="daterangeinputfield--error-end"
+              legend="Error example"
+              isReversed={isReversed}
+              inputRangeStartProps={inputRangeStartProps}
+              inputRangeEndProps={inputRangeEndProps}
+              locale={locale}
+              status={{
+                dateEnd: "error",
+              }}
+              validationMessage={
+                '"Date to" cannot be earlier than the "Date from" selection.'
+              }
+            />
+            <DateRangeInputField
+              id="daterangeinputfield--error-both"
+              legend="Error example"
+              isReversed={isReversed}
+              inputRangeStartProps={inputRangeStartProps}
+              inputRangeEndProps={inputRangeEndProps}
+              locale={locale}
+              status={{
+                dateEnd: "error",
+                dateStart: "error",
+              }}
+              validationMessage={
+                <ul style={{ margin: "0px", paddingInlineStart: "24px" }}>
+                  <li>
+                    "Date to" cannot be earlier than the "Date from" selection.
+                  </li>
+                  <li>"Date from" is not a valid date selection.</li>
+                </ul>
```

Comment:
Since this will be a consistent way of displaying multiple messages, I suggest this be a subcomponent where the consumer can just pass in: 
```
status={{
  dateEnd: "error",
  dateStart: "error",
}}
validationMessage={{
  dateStart: '"Date from" is close to the submission date.',
  dateEnd:
    '"Date to" cannot be earlier than the "Date from" selection.',
}}
```
and it renders the list for them.

Of course, we should still allow them to render whatever they want if they choose not to pass in the object but instead just a JSX.Element (which it seems you've already done).